bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java

```diff
@@ -85,20 +87,20 @@ public boolean equals(Object o) {
         TupleValueImpl that = (TupleValueImpl) o;
         return minSize == that.minSize &&
                 hasRestElement == that.hasRestElement &&
-                tupleType.equals(that.tupleType) &&
+                type.equals(that.type) &&
                 Arrays.equals(refValues, that.refValues);
     }
 
     @Override
     public int hashCode() {
-        int result = Objects.hash(tupleType, minSize, hasRestElement);
+        int result = Objects.hash(type, tupleType, minSize, hasRestElement);
         result = 31 * result + Arrays.hashCode(refValues);
         return result;
     }
 
     public TupleValueImpl(Object[] values, TupleType type) {
         this.refValues = values;
-        this.tupleType = type;
+        this.type = this.tupleType = type;
```

Comment:
Why we need this change?

Reply:
for `getType()` API it always returns the `this.type` field. So if we create a TupleValue with this constructor, we need to initialize the `type` field to avoid NPE.